tools/src/bin/sha256tree-benching.rs

```diff
@@ -0,0 +1,449 @@
+        // total nodes == (leaf_count * 2) - 1
+        samples_time_native.push((((leaf_count * 2) - 1) as f64, duration));
+        samples_cost_native.push((((leaf_count * 2) - 1) as f64, cost as f64));
+
+        // clvm
+        let start = Instant::now();
+        let red = run_program(a, &dialect, sha_prog, tree, 11_000_000_000).unwrap();
+        let cost = red.0;
+        let result_2 = node_to_bytes(a, red.1).expect("should work");
+        assert_eq!(result_1, result_2);
+        let duration = start.elapsed().as_nanos() as f64;
+        // subtract out the duration of hashing all the 32byte chunks
+        let duration = duration - ((2 * leaf_count - 1) as f64 * bytes32_clvm_time);
+        // subtract out the cost of hashing all the 32byte chunks
+        let cost = cost as f64 - ((2 * leaf_count - 1) as f64 * bytes32_clvm_cost);
+
+        writeln!(output_clvm_time, "{}\t{}", leaf_count, duration).unwrap();
+        writeln!(output_clvm_cost, "{}\t{}", leaf_count, cost).unwrap();
+
+        // internal node count == leaf_count - 1
+        // total nodes == (leaf_count * 2) - 1
+        samples_time_clvm.push((((leaf_count * 2) - 1) as f64, duration));
+        samples_cost_clvm.push((((leaf_count * 2) - 1) as f64, cost as f64));
+    }
+
+    (
+        linear_regression_of(&samples_time_native).unwrap().0,
+        linear_regression_of(&samples_time_clvm).unwrap().0,
+        linear_regression_of(&samples_cost_native).unwrap().0,
+        linear_regression_of(&samples_cost_clvm).unwrap().0,
+    )
+}
+
+// this function is for comparing the cost per 32byte chunk of hashing between the native and clvm implementation
+#[allow(clippy::type_complexity)]
+fn time_per_byte_for_atom(
+    a: &mut Allocator,
+    sha_prog: NodePtr,
+    mut output_native_time: impl Write,
+    mut output_clvm_time: impl Write,
+    mut output_native_cost: impl Write,
+    mut output_clvm_cost: impl Write,
+) -> (
+    f64,
+    f64, // time slopes
+    f64,
+    f64, // cost slopes
+) {
+    let mut samples_time_native = Vec::<(f64, f64)>::new();
+    let mut samples_time_clvm = Vec::<(f64, f64)>::new();
+    let mut samples_cost_native = Vec::<(f64, f64)>::new();
+    let mut samples_cost_clvm = Vec::<(f64, f64)>::new();
+    let dialect = ChiaDialect::new(ENABLE_SHA256_TREE);
+
+    let op_code = a.new_small_number(63).unwrap();
+    let quote = a.one();
+    let mut atom = vec![0xff; 10_000];
+
+    for i in 0..10_000 {
+        atom.extend(std::iter::repeat_n(((i % 89) + 10) as u8, 32));
+
+        let atom_node = a.new_atom(&atom).unwrap();
+        let args = a.new_pair(quote, atom_node).unwrap();
+        let call = a.new_pair(args, a.nil()).unwrap();
+        let call = a.new_pair(op_code, call).unwrap();
+
+        let checkpoint = a.checkpoint();
+
+        // native
+        let start = Instant::now();
+        let red = run_program(a, &dialect, call, a.nil(), 11_000_000_000).unwrap();
+        let cost = red.0;
```

Comment:
this cost seems like a circular dependency. This program is meant to *establish* this cost, not present what it happens to be set to right now. It really seems like something that easy to misunderstand. I don't know what it tells us here, but I do see how it can be confusing. I think it would be best to remove it, but at least it should have a comment explaining what it's for.

The main risk I see is that it, accidentally, is used to make it appear as if  a bad cost is good, because it matches this.